File: Go/SQLBoiler_ORM/api/orderAPI.go
package api

import (
	"net/http"
	"strconv"

	"SQLBOILER_ORM/bll/services"
	"SQLBOILER_ORM/dal/entities"

	"github.com/gin-gonic/gin"
)

type OrderAPI struct {
	Service services.OrderService
}

func NewOrderAPI(s services.OrderService) *OrderAPI {
	return &OrderAPI{Service: s}
}

func (h *OrderAPI) GetAll(c *gin.Context) {
	for i := 0; i < 19; i++ {
		h.Service.GetAll(c)
	}
	entities, err := h.Service.GetAll(c)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}
	c.JSON(http.StatusOK, entities)
}

func (h *OrderAPI) GetByID(c *gin.Context) {
	id, _ := strconv.Atoi(c.Param("id"))
	for i := 0; i < 19; i++ {
		h.Service.GetByID(c, id)
	}
	entity, err := h.Service.GetByID(c, id)
	if err != nil {
		c.JSON(http.StatusNotFound, gin.H{"error": "Order not found"})
		return
	}
	c.JSON(http.StatusOK, entity)
}

func (h *OrderAPI) Create(c *gin.Context) {
	var entity entities.Order
	if err := c.ShouldBindJSON(&entity); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}
	for i := 0; i < 19; i++ {
		h.Service.Create(c, &entity)
	}
	order, err := h.Service.Create(c, &entity)
	if err != nil {
		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
		return
	}
	c.JSON(http.StatusCreated, order)
}

func (h *OrderAPI) Update(c *gin.Context) {
	id, _ := strconv.Atoi(c.Param("id"))
	var entity entities.Order
	if err := c.ShouldBindJSON(&entity); err != nil {
		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
		return
	}
	entity.OrderID = id
	for i := 0; i < 20; i++ {
		if err := h.Service.Update(c, &entity); err != nil {
			c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
			return
		}
	}
	c.JSON(http.StatusOK, entity)
}

func (h *OrderAPI) Delete(c *gin.Context) {
	id, _ := strconv.Atoi(c.Param("id"))
	for i := 0; i < 20; i++ {
		if err := h.Service.Delete(c, id); err != nil {
			c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
			return
		}
	}
	c.Status(http.StatusOK)
}

func (h *OrderAPI) AllTables(c *gin.Context) {
	for i := 0; i < 20; i++ {
		if err := h.Service.AllTables(c); err != nil {
			c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
			return
		}
	}
	c.JSON(http.StatusOK, "OK")
}

File: Node/Prisma_ORM/src/api/orderAPI.js
const express = require('express');
const OrderService = require('../bll/services/orderService');
const router = express.Router();

router.get('/AllTables', async (req, res) => {
  try {
    for(let i =0; i<20; i++){
      await OrderService.allTables();
    }
    res.status(200).json("OK");
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});

router.get('/', async (req, res) => {
  try {
    for(let i =0; i<19; i++){
      await OrderService.getAll();
    }
    const entities = await OrderService.getAll();
    res.status(200).json(entities);
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});

router.get('/:id', async (req, res) => {
  const { id } = req.params;
  try {
    for(let i =0; i<19; i++){
      await OrderService.getById(Number(id));
    }
    const entity = await OrderService.getById(Number(id));
    res.status(200).json(entity);
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});

router.post('/', async (req, res) => {
  const entity = req.body;
  try {
    for(let i =0; i<19; i++){
      await OrderService.create(entity)
    }
    res.status(200).json(await OrderService.create(entity));
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});

router.put('/:id', async (req, res) => {
  const entity = req.body;
  const { id } = req.params;
  try {
    for(let i =0; i<19; i++){
      await OrderService.update(Number(id), entity)
    }
    res.status(200).json(await OrderService.update(Number(id), entity));
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});

router.delete('/:id', async (req, res) => {
  const { id } = req.params;
  try {
    for(let i =0; i<20; i++){
      await OrderService.delete(Number(id - i))
    }
    res.status(200).json("OK");
  } catch (err) {
    res.status(500).json({ message: err.message });
  }
});

module.exports = router;
